test/solidity/Facets/EcoFacet.t.sol

```diff
@@ -0,0 +1,492 @@
+            .selector;
+
+        addFacet(diamond, address(ecoFacet), functionSelectors);
+        ecoFacet = TestEcoFacet(address(diamond));
+        ecoFacet.addDex(ADDRESS_UNISWAP);
+        ecoFacet.setFunctionApprovalBySignature(
+            uniswap.swapExactTokensForTokens.selector
+        );
+        ecoFacet.setFunctionApprovalBySignature(
+            uniswap.swapTokensForExactETH.selector
+        );
+        ecoFacet.setFunctionApprovalBySignature(
+            uniswap.swapETHForExactTokens.selector
+        );
+
+        setFacetAddressInTestBase(address(ecoFacet), "EcoFacet");
+
+        bridgeData.bridge = "eco";
+        bridgeData.destinationChainId = 10;
+
+        // Set addToMessageValue for native token tests (ERC20 tests will override this)
+        addToMessageValue = NATIVE_SOLVER_REWARD;
+    }
+
+    function getValidEcoData(
+        bool isNative
+    ) internal view returns (EcoFacet.EcoData memory) {
+        // Calculate solver reward based on token type
+        uint256 solverReward = isNative
+            ? NATIVE_SOLVER_REWARD
+            : TOKEN_SOLVER_REWARD;
+
+        // Create mock encoded route data for all chains
+        bytes memory mockEncodedRoute = hex"0102030405060708090a0b0c0d0e0f10";
+
+        return
+            EcoFacet.EcoData({
+                receiverAddress: USER_RECEIVER,
+                nonEVMReceiver: "",
+                prover: address(0x1234),
+                rewardDeadline: uint64(block.timestamp + 2 days),
+                solverReward: solverReward,
+                encodedRoute: mockEncodedRoute // Required for all chains now
+            });
+    }
+
+    function initiateBridgeTxWithFacet(bool isNative) internal override {
+        EcoFacet.EcoData memory ecoData = getValidEcoData(isNative);
+
+        if (isNative) {
+            // For native: send bridge amount + native reward as msg.value
+            ecoFacet.startBridgeTokensViaEco{
+                value: bridgeData.minAmount + addToMessageValue
+            }(bridgeData, ecoData);
+        } else {
+            // For ERC20: No msg.value needed, tokens already approved
+            ecoFacet.startBridgeTokensViaEco(bridgeData, ecoData);
+        }
+    }
+
+    function initiateSwapAndBridgeTxWithFacet(
+        bool isNative
+    ) internal override {
+        EcoFacet.EcoData memory ecoData = getValidEcoData(isNative);
+
+        if (isNative) {
+            // Swapping to native: send swap input + native reward
+            uint256 msgValue = swapData.length > 0
+                ? swapData[0].fromAmount + addToMessageValue
+                : addToMessageValue;
+            ecoFacet.swapAndStartBridgeTokensViaEco{ value: msgValue }(
+                bridgeData,
+                swapData,
+                ecoData
+            );
+        } else {
+            // Swapping from native to ERC20: No additional msg.value needed
+            uint256 msgValue = swapData.length > 0
```

Comment:
since this the swap and... function ```initiateSwapAndBridgeTxWithFacet``` we should not have a case where swapData.length == 0, therefore this could be simplified:

```solidity
ecoFacet.swapAndStartBridgeTokensViaEco{ value: swapData[0].fromAmount }(
                bridgeData,
                swapData,
                ecoData
            );
````